QPPSF-7908 PCF Processing

saquino0827 opened the pull request:
### Information
- QPPSF-7908

### Changes proposed in this PR:
- Adds PCF conversion currently with 2020 measures data info. (Another ticket will handle updating this)
- Adds unit and integration testing of decoding, validating and encoding data.

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [x] Updated documentation (`README.md`, etc.) depending if the changes require it.

KyleApfel commented on converter/src/main/java/gov/cms/qpp/conversion/decode/ClinicalDocumentDecoder.java:
```diff
@@ -30,6 +30,7 @@
 	public static final String ENTITY_TYPE = "entityType";
 	public static final String MIPS_PROGRAM_NAME = "mips";
 	public static final String CPCPLUS_PROGRAM_NAME = "cpcPlus";
+	public static final String PCF = "PCF";
```
I thought talking with Michelle she was going to keep this all lowercase. Originally it was case insensitive so any caps was going to be accepted, but we talked her down to all lowercase. That said, I see the other variables not being appended with `_PROGRAM_NAME` are always in caps so might be some context I'm missing...

saquino0827 replied:
To be safe, I changed the conversion to use lowercasing as Conversion Tool's validation only ensures the value is PCF regardless of case sensitivity.

KyleApfel commented on converter/src/test/java/gov/cms/qpp/acceptance/PcfRoundTripTest.java:
```diff
@@ -0,0 +1,62 @@
+
+import static com.google.common.truth.Truth.assertThat;
+
+public class PcfRoundTripTest {
```
Reading the code, looks similar to what you were saying during sprint planning. Could you give a quick one-line comment here on the class? Don't think it's super necessary for the methods.

saquino0827 replied:
Went ahead and added comments for both negative and positive test cases.

KyleApfel reviewed (approved):
As far as I can tell everything looks good! Added one comment that might be due to my lack of domain knowledge, then a small blurb on comments where you think makes sense. There one class I knew how it functions only because of your breakdown during sprint planning, but could do with a comment on those.

saquino0827 commented:
> As far as I can tell everything looks good! Added one comment that might be due to my lack of domain knowledge, then a small blurb on comments where you think makes sense. There one class I knew how it functions only because of your breakdown during sprint planning, but could do with a comment on those.

Went ahead and added what you requested and fixed the build. Should be good to go now once this build runs.

KyleApfel reviewed (approved):
Thanks Sam!!
